scripts/lib/CIME/case/check_input_data.py

```diff
@@ -205,13 +244,57 @@ def check_input_data(case, protocol="svn", address=None, input_data_root=None, d
                             no_files_missing = False
 
                             if (download):
+                                if firstdownload:
+                                    # Get the md5 checksum file
+                                    got_chksum = _download_checksum_file(server, input_data_root)
+                                    firstdownload = False
+                                isdirectory=rel_path.endswith(os.sep)
                                 no_files_missing = _download_if_in_repo(server, input_data_root, rel_path.strip(os.sep),
-                                                               isdirectory=rel_path.endswith(os.sep))
+                                                                        isdirectory=isdirectory)
+                                if got_chksum and no_files_missing and not isdirectory:
+                                    verify_chksum(input_data_root,"inputdata_checksum.dat", rel_path.strip(os.sep))
                         else:
+                            if chksum:
+                                verify_chksum(input_data_root,"inputdata_checksum.dat", rel_path.strip(os.sep))
+                                logger.info("Chksum passed for file {}".format(os.path.join(input_data_root,rel_path)))
                             logging.debug("  Already had input file: '{}'".format(full_path))
 
+
+
                 else:
                     model = os.path.basename(data_list_file).split('.')[0]
                     logging.warning("Model {} no file specified for {}".format(model, description))
 
     return no_files_missing
+
+
+def verify_chksum(input_data_root, checksumfile, filename):
+    if not chksum_hash:
+        hashfile = os.path.join(input_data_root,checksumfile)
+        if not os.path.isfile(hashfile):
+            expect(False, "Failed to find or download file "+hashfile+
+                   " download from ftp:ftp.cgd.ucar.edu/cesm/inputdata_chksum.dat")
+
+        with open(hashfile) as fd:
+            lines = fd.readlines()
+            for line in lines:
+                lsplit = line.split()
+                if len(lsplit) < 8:
+                    continue
```

Comment:
Would it be feasible to require a header line on the checksum file, so that you could confirm that indeed the 1st column is the checksum and the 8th column is the file name?

I don't feel this is necessary, but it could help avoid some hard-to-debug errors in the future.

Reply:
I am working with an existing chksum file that Mark Moore generates - I didn't really want to make any request that he change the format of the file on his server.